bot/commands.ts

```diff
@@ -129,29 +133,87 @@ const commands: Record<string, Command> = {
 			contexts: [0, 2],
 		},
 		execute: async (interaction) => {
-			if (interaction?.guildId) {
-				const guild = interaction.guild?.id
-				const user = interaction.user.id
-				const xp = await makeGETRequest(guild as string, user)
-
-				if (!xp) {
-					await interaction.reply({
-						ephemeral: true,
-						content: "No XP data available."
-					});
-					return;
-				}
-
-				const progress = xp.user_progress_next_level;
-				const progressBar = createProgressBar(progress);
+			const guild = interaction.guild?.id
+			const user = interaction.user.id
+			const xp = await makeGETRequest(guild as string, user)
 
+			if (!xp) {
 				await interaction.reply({
-					embeds: [
+					ephemeral: true,
+					content: "No XP data available."
+				});
+				return;
+			}
+			
+			const card = new RankCardBuilder()
+  			.setDisplayName((interaction.member as GuildMember).displayName)
+        .setAvatar(interaction.user.displayAvatarURL()) // user avatar
+        .setCurrentXP(300) // current xp
+        .setRequiredXP(600) // required xp
+        .setLevel(2) // user level
+        .setRank(5) // user rank
+        .setOverlay(90) // overlay percentage. Overlay is a semi-transparent layer on top of the background
+        .setBackground("#23272a")
+			
+			if (interaction.user.discriminator !== "0") {
+			  card.setUsername("#" + interaction.user.discriminator)
+			} else {
+			  card.setUsername("@" + interaction.user.username)
+			}
+			
+      const color = await getColor(interaction.user.displayAvatarURL({ extension: "png" }));
+      card.setStyles({
+        progressbar: {
+          thumb: {
+            style: {
+              backgroundColor: `rgb(${color[0]}, ${color[1]}, ${color[2]})`
```

Comment:
Maybe in the future we could consider using the colour of their top role?

Reply:
that would probably be better yeah, I'll implement it